src/expressions/operators/arithmetic/BinaryEvaluationFunctionMap.ts

```diff
@@ -0,0 +1,238 @@
+import { ValueType } from '../../../expressions/dataTypes/Value';
+import {
+	addDuration as addDurationToDateTime,
+	subtract as dateTimeSubtract,
+	subtractDuration as subtractDurationFromDateTime,
+} from '../../../expressions/dataTypes/valueTypes/DateTime';
+import {
+	add as dayTimeDurationAdd,
+	divide as dayTimeDurationDivide,
+	divideByDayTimeDuration as dayTimeDurationDivideByDayTimeDuration,
+	multiply as dayTimeDurationMultiply,
+	subtract as dayTimeDurationSubtract,
+} from '../../../expressions/dataTypes/valueTypes/DayTimeDuration';
+import {
+	add as yearMonthDurationAdd,
+	divide as yearMonthDurationDivide,
+	divideByYearMonthDuration as yearMonthDurationDivideByYearMonthDuration,
+	multiply as yearMonthDurationMultiply,
+	subtract as yearMonthDurationSubtract,
+} from '../../../expressions/dataTypes/valueTypes/YearMonthDuration';
+
+/**
+ * A hash function that is used to create the keys for the ruleMap.
+ * @param left the ValueType of the left part of the operator
+ * @param right the ValueType of the right part of the operator
+ * @param op the operator
+ * @returns a number that can be used as a key
+ */
+export function hash(left: ValueType, right: ValueType, op: string): number {
+	return (
+		((left as number) << 20) +
+		((right as number) << 12) +
+		(op.charCodeAt(0) << 8) +
+		op.charCodeAt(1)
+	);
+}
+
+/**
+ * The map with every possible combination of operands.
+ * returns a the return type of the operation.
+ */
+export const returnTypeMap: { [key: number]: ValueType } = {
+	[hash(ValueType.XSNUMERIC, ValueType.XSNUMERIC, 'addOp')]: undefined,
```

Comment:
the return type here is `XSNUMERIC`, right?

Reply:
Yes and the types of these undefined cases can actually get covered by the `determineReturnType` function that we have expanded a few weeks ago so these lines can actually just get removed from the map. Did that in commit fdd04dccd8d0fc492acd401e4b27c53b4b724f00